Add updateWidth for inflections tab

irina060981 opened the pull request:
https://github.com/alpheios-project/components/issues/184

1) I have moved the code for calculate paddings to a separate method - `calcWidthPaddings`  and use it in `mounted` hook and in `setContentWidth` (if paddings are zero) - `panel.vue`

2) I have created `calcScrollPadding` and use it in `setContentWidth`  - `panel.vue`

3) I have created `updateWidth` method in `inflections.vue` with two variants of action - with timeout or not. Because we need to recalc width after rendering selected view and it could need some time for rendering in html structure and return correct `offsetWidth`

4) I have added `updateWidth` method calls to 
    a) change visibility event (it was there before)
    b) to change partOfSpeach - it calls render of selectedView
    c) to change viewSelectore - it calls render of selectedView
    d) and to show/hide additional data

kirlat commented on src/vue-components/inflections.vue:
```diff
@@ -311,6 +317,17 @@
     },
 
     methods: {
+      updateWidth: function (timeout) {
```
I'm wondering if timeout can be replaced with something else as I believe timeouts in situations like that can be unreliable. If we need a timeout to wait till a view is rendered by Vue, and it takes up to 400ms to render a simple view, it can take longer for more complex views. This will result in code to stop working on some views while working on others. It will also be a floating bug because it can render faster or slower depending on a machine and environment resulting in things being really hard to reproduce.

Are there any other solutions available? Can we use something like `Vue.nextTick()`?

irina060981 commented:
@kirlat, great notice! You are quite right! I have changed timeout to Vue.nextTick - nad it becomes more consistent.
Also I have added +1 to inflections width for cases when it has double format, because otherwise - offesetWidth returns interger value rounded down and at the end we havde a horizontal scroll bar.

kirlat commented on src/vue-components/inflections.vue:
```diff
@@ -170,7 +172,9 @@
-        sfCollapsed: true
+        sfCollapsed: true,
+
+        updateContentDelay: 400
```
Probably this is not needed any more and can be removed

kirlat commented on src/vue-components/inflections.vue:
```diff
@@ -296,7 +303,8 @@
         if (visibility) {
           // If container is become visible, update parent with its width
-          this.$emit('contentwidth', this.htmlElements.content.offsetWidth)
+          // this.$emit('contentwidth', this.htmlElements.content.offsetWidth)
```
Probably we can remove this comment too

irina060981 commented:
Thank you, @kirlat! I have fixed and code becomes cleaner :)
